src/main/java/com/microsoft/sqlserver/jdbc/SQLServerSpatialDatatype.java

```diff
@@ -760,9 +866,23 @@ protected void readPointWkt() throws SQLServerException {
 
             try {
                 coords[numOfCoordinates] = sign * new BigDecimal(wkt.substring(startPos, currentWktPos)).doubleValue();
+
+                if (numOfCoordinates == 2) {
+                    hasZvalues = true;
+                } else if (numOfCoordinates == 3) {
+                    hasMvalues = true;
+                }
             } catch (Exception e) { // modify to conversion exception
-                MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_illegalWKTposition"));
-                throw new SQLServerException(form.format(new Object[] {currentWktPos}), null, 0, null);
+                // handle NULL case
+                // the first check ensures that there is enough space for the wkt to have NULL
+                if (wkt.length() > currentWktPos + 3
+                        && wkt.substring(currentWktPos, currentWktPos + 4).equalsIgnoreCase("null")) {
+                    coords[numOfCoordinates] = Double.NaN;
+                    currentWktPos = currentWktPos + 4;
+                } else {
+                    MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_illegalWKTposition"));
```

Comment:
Please create a method `throwIllegalWKTposition`. There are 5 places where you construct the Exception.